### PR TITLE
fix(iceberg): Change how MapType are mapped to Avro to support complex Map key type

### PR DESCRIPTION
**Pull request by cccs-eric**
Iceberg maps support complex types for both keys and values, but the Avro spec only support `strings` as a key type.  This PR changes how we map an Iceberg `MapType` to an Avro map.  It now maps to an `array` of `struct` with a `key` and a `value` field, as suggested by @rslanka .

The UI will need to be adapted to render those to a map maybe?

### CODE REVIEW

File: metadata-ingestion/src/datahub/ingestion/source/iceberg/iceberg.py [hunk 1]
```diff
@@ -313,16 +313,30 @@ def _parse_datatype(type: IcebergTypes.Type, nullable: bool = False) -> Dict[str
...
+            ],
+        }
         return {
-            "type": "map",
-            "values": vt,
-            "native_data_type": str(map_type),
-            "key_type": kt,
-            "key_native_data_type": repr(map_type.key_type()),
+            "type": "array",
+            "items": map_entry,
```
Comment (gabe-lyons) [inline metadata-ingestion/src/datahub/ingestion/source/iceberg/iceberg.py:337]: What ultimately happens to this `items` field? It is not part of SchemaField.pdl, so is it transformed into another field later?

Comment (cccs-eric) [reply metadata-ingestion/src/datahub/ingestion/source/iceberg/iceberg.py:337]: To be honest, I don't know.  In the original Iceberg source PR, it got reviewed by a member of the Iceberg core team and one thing he pointed out was that Iceberg supports complex key types for map.  Here is the [original comment](https://github.com/datahub-project/datahub/pull/4431#discussion_r871663832) (and you can read the following comments from me and Ravindra).  I only implemented what was suggested to me, so I'm not sure what was the original plan for the UI.

Comment (rslanka) [reply metadata-ingestion/src/datahub/ingestion/source/iceberg/iceberg.py:337]: Hi @gabe-lyons, we have a few non pure AVRO schema types(hive, iceberg) that are leveraging our existing AVRO to SchemaField converter implementation. To get around the key-type limitation that is restricted to only the string type in AVRO for the other schema types, we will need to special-case this mechanism (array[__map_entry]) also as a map type in the UI soon. We can discuss more offline.

**Review by rslanka — APPROVED** [review]
LGTM!

**Pull request merged**